File: 020/problem020.exs
#!/usr/bin/env elixir
defmodule Problem020 do
  defp factorial(0), do: 1
  defp factorial(1), do: 1
  defp factorial(n), do: n * factorial(n - 1)

  defp sum_factorial_digits(x) do
    factorial(x)
    |> Integer.digits
    |> Enum.sum
  end

  def solve, do: sum_factorial_digits(100)
end

IO.puts Problem020.solve

File: 020/problem020.rb
#!/usr/bin/env ruby
class Integer
  def factorial
    return 1 if zero?
    return 1 if self == 1
    self * (self - 1).factorial
  end
end

# Ruby 2.4.2
def sum_factorial_digits(x)
  x.factorial.digits.reduce(:+)
end

# Ruby 2.2.3
# def sum_factorial_digits(x)
#   x.factorial.to_s.chars.map(&:to_i).reduce(:+)
# end

puts sum_factorial_digits(100)
